nav2_collision_monitor/src/collision_monitor_node.cpp

```diff
@@ -420,6 +424,10 @@ bool CollisionMonitor::processApproach(
 {
   polygon->updatePolygon();
 
+  if (!polygon->isShapeSet()) {
```

Comment:
It would seems sensible to me to have this before updating since we shouldn't need to update if its not set.

Reply:
The main thing here is that `updatePolygon()` optionally updating polygon by request by calling `getFootprintInRobotFrame()` FootprintSubscriber API. However, this might not update polygon, e.g. if FootprintSubscriber has not received footprint yet, or if we are using `Circle` object instead of `Polygon`.
For cases when `updatePolygon()` will set polygon shape, we need to check that shape is set after this call, in order to not miss the cycle when we already have footprint published, and thus it could be handled.
So, this check should be after `updatePolygon()` call for approach model.